app/knights/knight.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+
+class Knight:
+
+    def __init__(self, name: str, power: int, hp: int,
+                 armour: list[dict[str, int]],
+                 weapon: dict[str, int],
+                 potion: dict[str, dict[str, int]]) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+        self.apply_armour(armour)
+        self.apply_weapon(weapon)
+        self.apply_potion(potion)
+
+    def apply_armour(self,
+                     armour: list[dict[str, int]]) -> list[dict[str, int]]:
+        for part in armour:
+            if "protection" in part:
+                self.protection += part["protection"]
```

Comment:
You can simplify this:
```suggestion
            self.protection += part.get("protection", 0)
```

Reply:
The same with weapon

Reply:
time to repeat dict methods ((